src/components/button/Button.tsx

```diff
@@ -22,33 +21,58 @@ const textComponentMap = {
   large: UITextDisplay,
 };
 
-/** @type {React.ComponentType<any>} */
+interface Props {
```

Comment:
I think it should extend BoxProps, since it's using Box internally and spreading rest props on it. I just realised this. Need to do it myself as well for the Avatar component.